lib/internal/Magento/Framework/CompiledInterception/Generator/CompiledInterceptor.php

```diff
@@ -0,0 +1,399 @@
+    {
+        if ($parentConstructor == null) {
+            $parameters = [];
+            $body = [];
+        } else {
+            $parameters = $parentConstructor->getParameters();
+            foreach ($parameters as $parameter) {
+                $parentCallParams[] = '$' . $parameter->getName();
+            }
+            $body = ["parent::__construct(" . implode(', ', $parentCallParams) .");"];
+        }
+        foreach ($properties as $type => $name) {
+            $this->_classGenerator->addUse($type);
+            $this->classProperties[] = [
+                'name' => $name,
+                'visibility' => 'private',
+                'docblock' => [
+                    'tags' => [['name' => 'var', 'description' => substr(strrchr($type, "\\"), 1)]],
+                ]
+            ];
+        }
+        $extraParams = $properties;
+        $extraSetters = array_combine($properties, $properties);
+        foreach ($parameters as $parameter) {
+            if ($parameter->getType()) {
+                $type = $parameter->getType()->getName();
+                if (isset($properties[$type])) {
+                    $extraSetters[$properties[$type]] = $parameter->getName();
+                    unset($extraParams[$type]);
+                }
+            }
+        }
+        $parameters = array_map(array($this, '_getMethodParameterInfo'), $parameters);
+        /* foreach ($extraParams as $type => $name) {
+            array_unshift($parameters, [
+                'name' => $name,
+                'type' => $type
+            ]);
+        } */
+        foreach ($extraSetters as $name => $paramName) {
+            array_unshift($body, "\$this->$name = \$$paramName;");
+        }
+        foreach ($extraParams as $type => $name) {
+            array_unshift($body, "//TODO fix di in production mode");
+            array_unshift($body, "\$$name = \\Magento\\Framework\\App\\ObjectManager::getInstance()->get(\\$type::class);");
+        }
+
+        $this->classMethods[] = [
+            'name' => '__construct',
+            'parameters' => $parameters,
+            'body' => implode("\n", $body),
+            'docblock' => ['shortDescription' => '{@inheritdoc}'],
+        ];
+
+    }
+
+    protected function _getClassMethods()
+    {
+        return $this->classMethods;
+    }
+
+    protected function _getClassProperties()
+    {
+        return $this->classProperties;
+    }
+
+    private function addCodeSubBlock(&$body, $sub, $indent = 1)
+    {
+        foreach ($sub as $line) {
+            $body[] = str_repeat("\t", $indent) . $line;
+        }
+    }
+
+    /**
+     * @param \ReflectionMethod $method
+     * @param $conf
+     * @param $parameters
+     * @return array
+     * @SuppressWarnings(PHPMD.CyclomaticComplexity)
+     * @SuppressWarnings(PHPMD.NPathComplexity)
+     */
+    protected function _getMethodSourceFromConfig(\ReflectionMethod $method, $conf, $parameters, $returnVoid)
```

Comment:
all the new methods should be private and should follow new naming standards (should not start from the underscore)

Reply:
fixed